main/ZactuatorFASTLED.ino

```diff
@@ -0,0 +1,238 @@
+/*
+  OpenMQTTGateway - ESP8266 or Arduino program for home automation
+
+  This gateway enables to use RGBLED strips like WS2812
+
+    Copyright: (c)
+
+    This file is part of OpenMQTTGateway.
+
+    OpenMQTTGateway is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    OpenMQTTGateway is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "User_config.h"
+
+#ifdef ZactuatorFASTLED
+#include <ArduinoJson.h>
+
+#ifdef ESP8266
+#include <FastLED.h>
+#else
+#include <FastLED.h>
+#endif
+enum LEDState {OFF,FIRE,GENERAL};
+LEDState currentLEDState;
+
+CRGB leds[FASTLED_NUM_LEDS];
+CRGB ledColorBlink[FASTLED_NUM_LEDS];
+StaticJsonBuffer<200> jsonBuffer;
+bool blinkLED[FASTLED_NUM_LEDS];
+const long blinkInterval = 300; 
+const long fireUpdate = 20;
+unsigned long previousMillis = 0; 
+CRGBPalette16 gPal;
+
+void setupFASTLED() {
+  trc(F("FASTLED_DATA_PIN "));
+  trc(String(FASTLED_DATA_PIN));
+  trc(F("FASTLED_NUM_LEDS "));
+  trc(String(FASTLED_NUM_LEDS));
+  trc(F("ZactuatorFASTLED setup done "));
+  FastLED.addLeds<FASTLED_TYPE, FASTLED_DATA_PIN>(leds, FASTLED_NUM_LEDS);
+}
+
+
+
+void FASTLEDLoop()
+{
+   if (currentLEDState == GENERAL)
+   {
+    unsigned long currentMillis = millis();
+    if (currentMillis - previousMillis >= blinkInterval) 
+    {
+      // save the last time you blinked the LED
+      previousMillis = currentMillis;
+
+      for (int i = 0 ; i < FASTLED_NUM_LEDS; i++ ) 
+      {
+      
+        if (blinkLED[i])
+        {
+          CRGB currentcolor = leds[i];
+          switch (currentcolor)
+          {
+            case CRGB::Black:
+                leds[i] =ledColorBlink[i];
+            break;
+
+            default:
+              ledColorBlink[i] = leds[i];
+                    leds[i] = CRGB::Black;  
```

Comment:
can you use a formatter, to fix the indentation of the code. currently it's hard to read the code.

Reply:
I will do a code format.